extensions/amp-analytics/0.1/visibility-manager.js

```diff
@@ -376,9 +378,27 @@ export class VisibilityManager {
           'elementHeight': layoutBox.height,
         }));
       }
+
+      // If spec includes initial scroll depth, set it
+      if (spec['initialScrollDepth']) {
+        model.maybeSetInitialScrollDepth(viewport.getScrollTop());
+        state['initialScrollDepth'] = model.getInitialScrollDepth();
+      }
+
+      if (spec['maxScrollDepth']) {
+        state['maxScrollDepth'] = model.getMaxScrollDepth();
+      }
+
       callback(state);
     });
 
+    // If spec includes max scroll depth, add scroll listener
+    if (spec['maxScrollDepth']) {
+      viewport.onScroll(() => {
```

Comment:
Please use the `viewport.onChange` as we agreed. It throttles a bit more, but I believe we agreed that's acceptable.

Reply:
Done